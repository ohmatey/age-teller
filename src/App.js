import React from 'react'
import { Form, FormControl } from 'react-bootstrap'
import { compose, withState, withHandlers } from 'recompose'

import "./App.css"
import AgeStats from './AgeStats'

export const App = ({ handleDateChange, birthDate }) => (
  <div className="App">
    <h1>Enter your birthday!</h1>
    <Form inline>
      <FormControl
        type="date"
        onChange={handleDateChange}
        value={birthDate}
      />
    </Form>
    <AgeStats date={birthDate} />
  </div>
)

const enhance = compose(
  withState('birthDate', 'changeDate', ''),
  withHandlers({
    handleDateChange: ({ changeDate }) => e => changeDate(e.target.value)
  })
)

export default enhance(App)
