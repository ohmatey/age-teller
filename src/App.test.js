import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { App } from './App'

test('App component renders without props', () => {
  const component = renderer.create(
    <App />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('App component renders with date', () => {
  const component = renderer.create(
    <App date={'24-11-2015'} />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('App renders date', () => {
  const mockBirthDate = '24-11-2015'
  const component = shallow(
    <App birthDate={mockBirthDate} />
  )

  expect(component.find('FormControl').props().value).toEqual(mockBirthDate)
})

test('App renders date', () => {
  const mockBirthDate = '24-11-2015'
  const component = shallow(
    <App birthDate={mockBirthDate} />
  )

  expect(component.find('FormControl').props().value).toEqual(mockBirthDate)
})

test('App handleDateChange works', () => {
  const mockHandleChange = jest.fn()
  const component = shallow(
    <App handleDateChange={mockHandleChange} />
  )

  component.instance().props.handleDateChange()
  expect(mockHandleChange).toHaveBeenCalled()
})
