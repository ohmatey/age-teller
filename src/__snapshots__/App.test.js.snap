// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders with date 1`] = `
<div
  className="App"
>
  <h1>
    Enter your birthday!
  </h1>
  <form
    className="form-inline"
  >
    <input
      className="form-control"
      id={undefined}
      onChange={undefined}
      type="date"
      value={undefined}
    />
  </form>
  <div>
    <h2 />
  </div>
</div>
`;

exports[`App component renders without props 1`] = `
<div
  className="App"
>
  <h1>
    Enter your birthday!
  </h1>
  <form
    className="form-inline"
  >
    <input
      className="form-control"
      id={undefined}
      onChange={undefined}
      type="date"
      value={undefined}
    />
  </form>
  <div>
    <h2 />
  </div>
</div>
`;
